File: OC.Domain/Models/Branches/BranchLeadership.cs
using OC.Domain.Interfaces;
using OC.Domain.Models.Members;

namespace OC.Domain.Models.Branches
{
    public class BranchLeadership : AuditableEntity, IEntityBase
    {
        public long Id { get; set; }
        public long BranchId { get; set; }
        public long LeadershipRoleId { get; set; }
        public long MemberId { get; set; }
        public bool IsActive { get; set; }

        // Relationships
        public virtual Branch Branch { get; set; }
        public virtual LeadershipRole LeadershipRole { get; set; }
        public virtual Member Member { get; set; }
    }
}

File: OC.UI/src/app/blci/api/branches/branch-leadership.ts
import { LeadershipRole } from "../members/leadership-role";
import { Member } from "../members/member";
import { Branch } from "./branch";

export interface BranchLeadership {
    id: number;
    branchId: number;
    leadershipRoleId: number;
    memberId: number;
    isActive: boolean;
    branch: Branch;
    leadershipRole: LeadershipRole;
    member: Member;
}
